src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -986,6 +1016,88 @@ public override BoundNode VisitIsPatternExpression(BoundIsPatternExpression node
             }
 
             return node;
+
+            // Returns `true` if the pattern only matches if the top-level input is not null.
+            // Returns `false` if the pattern only matches if the top-level input is null.
+            // Otherwise, returns `null`.
+            static bool? isTopLevelNullTest(BoundPattern pattern)
```

Comment:
the local function probably needs a negation (either changing its name, or changing return values).

Reply:
Done